Realm/ObjectServerTests/RLMPermissionsAPITests.m

```diff
@@ -871,56 +831,79 @@ - (void)disabled_testSortingPermissionsOnDate {
     XCTAssertLessThan([n2.updatedAt timeIntervalSinceReferenceDate], [n3.updatedAt timeIntervalSinceReferenceDate]);
 }
 
-- (void)disabled_testPermissionResultsIndexOfObjectWithPredicate {
+- (void)testPermissionResultsIndexOfObjectWithPredicate {
     // Get a reference to the permission results.
     XCTestExpectation *ex = [self expectationWithDescription:@"Get permission results."];
     __block RLMResults<RLMSyncPermission *> *results = nil;
-    [self.userA retrievePermissionsWithCallback:^(RLMResults<RLMSyncPermission *> *r, NSError *error) {
+    [self.userB retrievePermissionsWithCallback:^(RLMResults<RLMSyncPermission *> *r, NSError *error) {
         XCTAssertNil(error);
         XCTAssertNotNil(r);
         results = r;
         [ex fulfill];
     }];
     [self waitForExpectationsWithTimeout:2.0 handler:nil];
 
-    // Open a Realm
-    NSURL *url = REALM_URL();
-    __attribute__((objc_precise_lifetime)) RLMRealm *realm = [self openRealmForURL:url user:self.userA];
     NSString *uB = self.userB.identity;
+    // Open a Realm
+    {
+        NSURL *url = CUSTOM_REALM_URL(@"r1");
+        __attribute__((objc_precise_lifetime)) RLMRealm *realm = [self openRealmForURL:url user:self.userA];
+
+        // Give user B read permission for the Realm.
+        XCTestExpectation *ex = [self expectationWithDescription:@"Setting r1 permission for user B should work."];
+        [self.userA applyPermission:[[RLMSyncPermission alloc] initWithRealmPath:[makeTildeSubstitutedURL(url, self.userA) path]
```

Comment:
I think this is `APPLY_PERMISSION_WITH_MESSAGE`?

Reply:
Yes it is. Thank you for catching that.